react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+export default class ValidateInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.state = { password: '', fontAwesomeClass: 'fa-solid fa-xmark' };
+    this.passwordText = 'A password is required.';
+  }
+
+  handlePasswordChange(event) {
+    this.setState({ password: event.target.value });
+    if (event.target.value.length === 0) {
+      this.setState({ fontAwesomeClass: 'fa-solid fa-xmark' });
+      this.passwordText = 'A password is required.';
+    }
+    if (event.target.value.length < 8 && event.target.value.length > 0) {
+      this.setState({ fontAwesomeClass: 'fa-solid fa-xmark' });
+      this.passwordText = 'Your password is too short.';
+    }
+    if (event.target.value.length >= 8) {
+      this.setState({ fontAwesomeClass: 'fa-solid fa-check' });
+      this.passwordText = '';
```

Comment:
Try to avoid adding extra stuff to `this`. You can put the error message in state. Instead of putting the class in state, conditionally choose it in `render()`.